### ISSUE TITLE
[Issue] mixnode: support listening on both ip4 and ip6 on OpenBSD

### ISSUE THREAD

**Issue by timkuijsten** [nymtech/nym#1781]
**Describe the issue**
It is currently not possible to let the mixnode listen to both IPv4 and IPv6 packets on OpenBSD, because only one `listening_address` can be configured and OpenBSD does not support IPv4 mapped addresses on a AF_INET6 socket for security reasons (`listening_address = '::'` will never accept ipv4 traffic, see [inet6(4)](https://man.openbsd.org/inet6.4)).

**Expected behaviour**
Possibility to specify multiple listening addresses in the mixnode config, or support `*` and let it expand to two socket addresses, namely the ipv4 wildcard `0.0.0.0` and the ipv6 wildcard `::`.

**Which area of Nym were you using?**
 - Application: mixnode
 - OS: OpenBSD
 - Version: nym binary(1.1.0)

**Additional context**
I've created a simple PoC so that in addation to my statically configured ipv6 address, an extra socket is created for the IPv4 wildcard:
```diff
commit 3d39f25b2c135cac80dff711187f1fde15a4142c (HEAD)
Author: Tim Kuijsten <info+git@netsend.nl>
Date:   Sun Nov 20 18:49:27 2022 +0100

    PoC let mixnode listen on ip4 and ip6

diff --git a/mixnode/src/node/mod.rs b/mixnode/src/node/mod.rs
index 0bca9b31..86c21728 100644
--- a/mixnode/src/node/mod.rs
+++ b/mixnode/src/node/mod.rs
@@ -23,7 +23,7 @@ use log::{error, info, warn};
 use mixnode_common::verloc::{self, AtomicVerlocResult, VerlocMeasurer};
 use rand::seq::SliceRandom;
 use rand::thread_rng;
-use std::net::SocketAddr;
+use std::net::{IpAddr, Ipv4Addr, SocketAddr};
 use std::process;
 use std::sync::Arc;
 use task::{wait_for_signal, ShutdownListener, ShutdownNotifier};
@@ -185,7 +185,13 @@ impl MixNode {
             self.config.get_mix_port(),
         );
 
-        Listener::new(listening_address, shutdown).start(connection_handler);
+        let listening_address2 = SocketAddr::new(
+            IpAddr::V4(Ipv4Addr::new(0, 0, 0, 0)),
+            self.config.get_mix_port(),
+        );
+
+        Listener::new(listening_address, shutdown.clone()).start(connection_handler.clone());
+        Listener::new(listening_address2, shutdown.clone()).start(connection_handler.clone());
     }
 
     fn start_packet_delay_forwarder(
```

This fixes at least the mixnode port test on [explorers.guru](https://mixnet.explorers.guru/mixnode/FS3uvkS7UkAcDJxXRf1ptpvx7jZT55SEeCNv5eiuKuQV) but I'm not sure if this is the right approach and if we want to do the same for the http and verloc ports.

**Comment by sven-hash** [comment]
Listen to IPv4 and v6 is not necessary you only need to be able to route IPv6 traffic

**Comment by timkuijsten** [comment]
Would it matter if I would only listen on ip6 instead of ip4? (and route of course to both ip4 and ip6)

**Comment by sven-hash** [comment]
Not it doesn't really matter, watch your routing score if it's increasing it means it works

**Comment by timkuijsten** [comment]
Note: I've tested running the node on a static IPv6 address and the routing score displayed on mixnet.explorers.guru was highly volatile and changed between 50% and 100% per hour which I find quite remarkable given the fact I've been running more servers behind this router (both v4 and v6) and can not reproduce any connection issues from outside. Because of this I've switched back to binding to an ip4 address and the score is now more stable (in the 94-100% range). I could only find one other mixnode advertising an ip6 address and it seems to have similar issues. I suspect the monitoring side to have issues monitoring ip6 addresses, but I could be wrong.
